yt_dlp/extractor/vanillo.py

```diff
@@ -0,0 +1,386 @@
+                http_code = 404
+            if http_code == 404:
+                self.raise_login_required(
+                    'Video not found or is private. '  # maybe use report_login here?
+                    'Sign in if you have access to this video. Use --cookies or --cookies-from-browser option',
+                    method=None,
+                )
+            elif http_code == 403:
+                raise ExtractorError(
+                    'Your Internet provider is likely blocked. Try again with another ISP or use VPN',
+                    expected=True,
+                )
+            raise
+
+        if video_info.get('status') != 'success':
+            raise ExtractorError('Video info API returned an error', expected=True)
+        data = video_info.get('data', {})
+        title = data.get('title') or video_id
+        description = data.get('description')
+        thumbnail = data.get('thumbnail')
+
+        uploader = data.get('uploader', {})
+        uploader_url = uploader.get('url')
+
+        # 2) Fix the ISO8601 date to remove leftover data
+        upload_date_raw = data.get('publishedAt')
+        upload_date = None
+        if upload_date_raw:
+            # Remove fractional seconds and any extra data after 'Z'
+            upload_date_raw = re.sub(r'\.\d+', '', upload_date_raw)
+            upload_date_raw = re.sub(r'Z.*$', 'Z', upload_date_raw)
+            try:
+                parsed_date = datetime.datetime.fromisoformat(upload_date_raw.replace('Z', '+00:00'))
+                upload_date = parsed_date.strftime('%Y%m%d')
+            except ValueError:
+                pass
+
+        duration = data.get('duration')
+
+        # Convert numeric fields safely
+        def safe_int(val):
+            try:
+                return int(val)
+            except (TypeError, ValueError):
+                return None
+
+        view_count = safe_int(data.get('views'))
+        comment_count = safe_int(data.get('totalComments'))
+        like_count = safe_int(data.get('likes'))
+        dislike_count = safe_int(data.get('dislikes'))
+
+        average_rating = None
+        if like_count is not None and dislike_count is not None:
+            total = like_count + dislike_count
+            if total > 0:
+                average_rating = round((like_count / total) * 5, 1)
+
+        categories = data.get('category')
+        if categories and not isinstance(categories, list):
+            categories = [categories]
+        tags = data.get('tags')
+
+        # 3) Get watch token (required for accessing manifests)
+        watch_token_url = 'https://api.vanillo.tv/v1/watch'
+        post_data = json.dumps({'videoId': video_id}).encode('utf-8')
+        watch_token_resp = self._download_json(
+            watch_token_url, video_id,
+            note='Downloading watch token',
+            data=post_data,
+            headers={'Content-Type': 'application/json'},
```

Comment:
inline url and data in the function call. you can just use `.encode()` since `utf-8` is the default